hazelcast-spring/src/main/java/com/hazelcast/spring/HazelcastConfigBeanDefinitionParser.java

```diff
@@ -617,8 +618,8 @@ public void handleJoin(Node node, BeanDefinitionBuilder networkConfigBuilder) {
                     handleMulticast(child, joinConfigBuilder);
                 } else if ("tcp-ip".equals(name)) {
                     handleTcpIp(child, joinConfigBuilder);
-                } else if ("aws".equals(name)) {
-                    handleAws(child, joinConfigBuilder);
+                } else if (AliasedDiscoveryConfigUtils.supports(name)) {
```

Comment:
Missing in `com.hazelcast.config.XmlConfigBuilder#handleWanPublisherConfig`. You can add a test for it as well.